lightning/src/routing/gossip.rs

```diff
@@ -628,15 +628,97 @@ impl fmt::Display for ChannelUpdateInfo {
 	}
 }
 
-impl_writeable_tlv_based!(ChannelUpdateInfo, {
-	(0, last_update, required),
-	(2, enabled, required),
-	(4, cltv_expiry_delta, required),
-	(6, htlc_minimum_msat, required),
-	(8, htlc_maximum_msat, required),
-	(10, fees, required),
-	(12, last_update_message, required),
-});
+impl Writeable for ChannelUpdateInfo {
+	fn write<W: ::util::ser::Writer>(&self, writer: &mut W) -> Result<(), io::Error> {
+		write_tlv_fields!(writer, {
+			(0, self.last_update, required),
+			(2, self.enabled, required),
+			(4, self.cltv_expiry_delta, required),
+			(6, self.htlc_minimum_msat, required),
+			(8, Some(self.htlc_maximum_msat), required),
+			(10, self.fees, required),
+			(12, self.last_update_message, required),
+		});
+		Ok(())
+	}
+}
+
+impl MaybeReadable for ChannelUpdateInfo {
+	fn read<R: io::Read>(reader: &mut R) -> Result<Option<Self>, DecodeError> {
+		macro_rules! read_channel_update_tlv_fields {
+			($reader: expr, {$(($type: expr, $field: ident)),* $(,)*}) => {{
```

Comment:
I agree that this is generally a pretty ugly way to go about it and am open to suggestions. However, I think just having a wrapper call `read` and catch the error result won't work, because we still have to read/write `htlc_maximum_msat` as an `Option<u64>` to remain backwards compatible?

Reply:
Yep, after clarifying I went ahead and implemented the wrapper idea with c554e641cebd9db0ab0892114f1cd4d79d44325c.